modules/pipeline/providers/leaderworker/listen.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leaderworker
+
+import (
+	"context"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+)
+
+func (p *provider) ListenPrefix(ctx context.Context, prefix string, putHandler, deleteHandler func(context.Context, *clientv3.Event)) {
+	for func() bool {
```

Comment:
too complex func, can you separate to some clear meaning small funcs?

Reply:
no

Reply:
simple enough.

Reply:
it's exactly clearly enough if you open this func in text editor. only thirty lines to handle continuous etcd prefix-key listen properly.
`for` is used in two ways: continuous-listen-loop and range arrays.
the outermost layer for-loop handles a `func() bool` to avoid continue from multi-layer loop(you need to use named loop).

Reply:
removed outermost layer of loop.
there is no issues of continue from multi-layer loop in this code.